compaction/compaction_manager.cc

```diff
@@ -1127,12 +1127,21 @@ future<> compaction_manager::drain() {
 }
 
 future<> compaction_manager::stop() {
-    do_stop();
-    if (auto cm = std::exchange(_task_manager_module, nullptr)) {
-        co_await cm->stop();
-    }
-    if (_state != state::none) {
-        co_return co_await std::move(*_stop_future);
+    try {
+        do_stop();
+        if (auto cm = std::exchange(_task_manager_module, nullptr)) {
+            auto f = cm->stop();
+            if (!_stop_future) {
+                _stop_future = std::move(f);
+            } else {
+                _stop_future = when_all(std::move(*_stop_future), std::move(f)).discard_result();
+            }
+        }
+        if (_state != state::none) {
+            co_return co_await std::move(*_stop_future);
+        }
+    } catch (...) {
+        cmlog.error("Failed to stop the manager: {}", std::current_exception());
```

Comment:
"eliminating" all exceptions (with `...`) may be useful to prevent crashes if we assume that the system is in a stable enough state to continue.
Here we don't handle the exception, we simply report about it.
I don't know if there is something better to do, maybe just start report all these `...` catches with some grepable text for later investigation.

Reply:
we have nothing better to do other than reporting the failure.
we can restrict catch to particular errors but I'm not sure it's worth it